src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -1428,6 +1428,14 @@ void MacroAssembler::check_klass_subtype_fast_path(Register sub_klass,
     ldrw(temp_reg, super_check_offset_addr);
     super_check_offset = RegisterOrConstant(temp_reg);
   }
+
+  // Don't check secondary_super_cache
+  if (super_check_offset.is_register()
```

Comment:
Do you see any effects from this particular change?

It adds a runtime check on the fast path for all subtype checks (irrespective of whether it checks primary or secondary super). Moreover, the very same check is performed after primary super slot is checked.

Unless `_secondary_super_cache` field is removed, unconditionally checking the slot at `super_check_offset` is benign.

Reply:
BTW `MacroAssembler::check_klass_subtype_fast_path` deserves a cleanup: `super_check_offset` can be safely turned into `Register` thus eliminating the code guarded by `super_check_offset.is_register() == false`.

Reply:
> Do you see any effects from this particular change?
> 
> It adds a runtime check on the fast path for all subtype checks (irrespective of whether it checks primary or secondary super). Moreover, the very same check is performed after primary super slot is checked.

OK. I think this was more for testing, but you make sense.